Mathlib/AlgebraicGeometry/Morphisms/Basic.lean

```diff
@@ -68,16 +68,76 @@ noncomputable section
 
 namespace AlgebraicGeometry
 
+section IsLocalAtTarget
+
+/--
+We say that `P : MorphismProperty Scheme` is local at the target if
+1. `P` respects isomorphisms.
+2. If `P` holds for `f : X ⟶ Y`, then `P` holds for `f ∣_ U` for any `U`.
+3. If `P` holds for `f ∣_ U` for an open cover `U` of `Y`, then `P` holds for `f`.
+-/
+class IsLocalAtTarget (P : MorphismProperty Scheme) : Prop where
+  /-- `P` respects isomorphisms. -/
+  respectsIso : P.RespectsIso
+  /-- If `P` holds for `f : X ⟶ Y`, then `P` holds for `f ∣_ U` for any `U`. -/
+  restrict : ∀ {X Y : Scheme} (f : X ⟶ Y) (U : Opens Y), P f → P (f ∣_ U)
+  /-- If `P` holds for `f ∣_ U` for an open cover `U` of `Y`, then `P` holds for `f`.  -/
+  of_sSup_eq_top :
+    ∀ {X Y : Scheme.{u}} (f : X ⟶ Y) (Us : Set (Opens Y)), sSup Us = ⊤ →
+      (∀ U ∈ Us, P (f ∣_ U)) → P f
```

Comment:
I also like the `OpenCover` version, but because of universe issue, we would need to show `OpenCover.{u} => restrict open => OpenCover.{v}` for both of these conditions.
If you think the `OpenCover` version is better I can change it.

Reply:
I think I prefer the `OpenCover` version while adding the set-theoretic versions as alternative constructors.